llvm/lib/IR/Operator.cpp

```diff
@@ -37,6 +37,10 @@ bool Operator::hasPoisonGeneratingFlags() const {
     // Note: inrange exists on constexpr only
     return GEP->isInBounds() || GEP->getInRangeIndex() != std::nullopt;
   }
+  case Instruction::ZExt:
+    if (auto *NNI = dyn_cast<PossiblyNonNegInst>(this))
```

Comment:
Why do we need a dyn_cast instead of a cast?

Reply:
This is in case it's a constant expression. I haven't completely finished the zext expr removal yet.